File: app/src/main/java/com/anthonymandra/rawdroid/data/entity/PathEntity.java
package com.anthonymandra.rawdroid.data.entity;

import android.arch.persistence.room.ColumnInfo;
import android.arch.persistence.room.PrimaryKey;
import android.provider.BaseColumns;

public abstract class PathEntity
{
	public static final String _ID = BaseColumns._ID;
	public static final String PATH = "path";
	public static final String DEPTH = "depth";
	public static final String PARENT = "parent";

	@PrimaryKey
	@ColumnInfo(name = _ID)
	public long id;

	@ColumnInfo(name = PATH)
	public String path;

	public long parent = -1;

	@ColumnInfo(name = DEPTH)
	public int depth;
}

File: app/src/main/java/com/anthonymandra/curator/data/PathEntity.kt
package com.anthonymandra.curator.data

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity
abstract class PathEntity {
    /**
     * Primary key
     */
    @PrimaryKey(autoGenerate = true)
    open var id: Long = 0

    /**
     * Path identifying the hierarchy of data
     */
    open var path: String = ""

    /**
     * This item's parent id or null if it is a root
     */
    open var parent: Long = -1

    /**
     * The depth at which this item resides (number of parents)
     */
    open var depth: Int = 0

    fun hasParent() = parent > 0
}
